src/oss-find-squats/FindSquatsTool.cs

```diff
@@ -0,0 +1,145 @@
+
+
+            SarifLog sarifLog = new SarifLog();
+            sarifLog.Version = SarifVersion.Current;
+            Run runItem = new Run();
+            runItem.Tool = new Tool();
+
+            runItem.Tool.Driver = new ToolComponent();
+            if (Assembly.GetEntryAssembly() is Assembly entryAssembly)
+            {
+                runItem.Tool.Driver.Name = entryAssembly.GetName().Name;
+
+                runItem.Tool.Driver.FullName = entryAssembly.GetCustomAttribute<AssemblyProductAttribute>()?
+                                                     .Product;
+
+                runItem.Tool.Driver.Version = entryAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?
+                                                    .InformationalVersion;
+            }
+
+            var descriptors = new List<ReportingDescriptor>();
+
+            foreach (var mutation in gen.Mutations)
+            {
+                ReportingDescriptor sarifRule = new ReportingDescriptor();
+                sarifRule.Name = mutation.Method.Name;
+                sarifRule.DefaultConfiguration = new ReportingConfiguration() { Enabled = true };
+            }
+
+            runItem.Tool.Driver.Rules = descriptors;
+
+            var sarifResults = new List<Result>();
+
+            foreach (var target in opts.Targets ?? Array.Empty<string>())
+            {
+                var purl = new PackageURL(target);
+                if (purl.Name is not null && purl.Type is not null)
+                {
+                    var manager = ProjectManagerFactory.CreateProjectManager(purl, null);
+                    if (manager is not null)
+                    {
+                        var mutationsDict = gen.Mutate(purl.Name);
+
+                        foreach ((var candidate, var rules) in mutationsDict)
+                        {
+                            Logger.Info($"Checking {candidate}.");
+                            var candidatePurl = new PackageURL(purl.Type, candidate);
+                            var versions = await manager.EnumerateVersions(candidatePurl);
+
+                            if (versions.Any())
+                            {
+                                Logger.Warn($"{candidate} package exists. Potential squat.");
+                                Result res = new Result();
+                                res.Locations = new List<Location>();
+                                Location loc = new Location();
+
+                                loc.LogicalLocation = new LogicalLocation()
+                                {
+                                    FullyQualifiedName = purl.ToString()
+                                };
+                                res.Locations.Add(loc);
+                                var rulesString = string.Join(',', rules);
+                                var message = $"Potential Squat candidate {candidate} detected. Generated by {rulesString}";
```

Comment:
This would do well as helper functions in SarifOutputBuilder. One reason I dont reference any sarif artifacts in the tools is that its one less reference to add. Only the shared project references the sarif nuget. This is not a requirement or anything, just the logic behind the helper class.

Reply:
I've added the rules that apply as tags but they don't show up in the sarif writer output.  Is that intentional?

Reply:
didnt understand this part...I dont see any tags here.

Reply:
I swear I did this. Pushing a new version in a couple minutes.

Reply:
Code for tags somehow didn't get committed before. Its working now.